zerver/lib/bugdown/__init__.py

```diff
@@ -734,40 +746,18 @@ def dropbox_image(self, url: str) -> Optional[Dict[str, Any]]:
             return image_info
         return None
 
-    def youtube_id(self, url: str) -> Optional[str]:
-        if not self.markdown.image_preview_enabled:
-            return None
-        # Youtube video id extraction regular expression from https://pastebin.com/KyKAFv1s
-        # Slightly modified to support URLs of the forms
-        #   - youtu.be/<id>
-        #   - youtube.com/playlist?v=<id>&list=<list-id>
-        #   - youtube.com/watch_videos?video_ids=<id1>,<id2>,<id3>
-        # If it matches, match.group(2) is the video id.
-        schema_re = r'(?:https?://)'
-        host_re = r'(?:youtu\.be/|(?:\w+\.)?youtube(?:-nocookie)?\.com/)'
-        param_re = r'(?:(?:(?:v|embed)/)|' + \
-            r'(?:(?:(?:watch|playlist)(?:_popup|_videos)?(?:\.php)?)?(?:\?|#!?)(?:.+&)?v(?:ideo_ids)?=))'
-        id_re = r'([0-9A-Za-z_-]+)'
-        youtube_re = r'^({schema_re}?{host_re}{param_re}?)?{id_re}(?(1).+)?$'
-        youtube_re = youtube_re.format(schema_re=schema_re, host_re=host_re, id_re=id_re, param_re=param_re)
-        match = re.match(youtube_re, url)
-        # URLs of the form youtube.com/playlist?list=<list-id> are incorrectly matched
-        if match is None or match.group(2) == 'playlist':
-            return None
-        return match.group(2)
-
     def youtube_title(self, extracted_data: Dict[str, Any]) -> Optional[str]:
         title = extracted_data.get("title")
         if title is not None:
             return "YouTube - {}".format(title)
         return None
 
-    def youtube_image(self, url: str) -> Optional[str]:
-        yt_id = self.youtube_id(url)
-
-        if yt_id is not None:
-            return "https://i.ytimg.com/vi/%s/default.jpg" % (yt_id,)
-        return None
+    def resize_youtube_image(self, extracted_data: Dict[str, Any]) -> None:
```

Comment:
`resize` seems a little mis-leading, since we are just changing the URL, and not actually "resizing".